Sources/FluentUI_iOS/Components/Pill Button Bar/SwiftUI/PillButtonView.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+#if canImport(FluentUI_common)
+import FluentUI_common
+#endif
+import SwiftUI
+
+/// A `PillButtonView` is a button in the shape of a pill.
+public struct PillButtonView: View, TokenizedControlView {
+    public typealias TokenSetKeyType = PillButtonTokenSet.Tokens
+    @ObservedObject public var tokenSet: PillButtonTokenSet
+
+    /// Initializes a `PillButtonView`.
+    ///
+    /// - Parameters:
+    ///   - style: The style of the pill button.
+    ///   - viewModel: The view model used to create the pill button.
+    ///   - action: The action performed when the pill button is tapped.
+    public init(style: PillButtonStyle,
+                viewModel: PillButtonViewModel,
+                action: (() -> Void)?) {
+        self.tokenSet = PillButtonTokenSet(style: { style })
+        self.action = action
+        self.viewModel = viewModel
+        self.isSelected = false
+        self.title = viewModel.title
+        self.accessibilityLabelWithUnreadDot = String(format: "Accessibility.TabBarItemView.UnreadFormat".localized, title)
+    }
+ 
+    public var body: some View {
+        tokenSet.update(fluentTheme)
+
+        return SwiftUI.Button {
+            action?()
+
+            if viewModel.isUnread {
+                viewModel.isUnread = false
+            }
+        } label: {
+            label()
+        }
+        .buttonStyle(PillButtonViewStyle(isSelected: isSelected,
+                                         isUnread: viewModel.isUnread,
+                                         tokenSet: tokenSet))
+        .accessibilityAddTraits(isSelected ? .isSelected : [])
+        .accessibilityLabel(viewModel.isUnread ? accessibilityLabelWithUnreadDot : title)
+        .showsLargeContentViewer(text: title)
+    }
+
+    @ViewBuilder
+    private func label() -> some View {
```

Comment:
Suggestion: could we move the label layout to a custom `LabelStyle` that could then be applied in the `PillButtonViewStyle`? Then this class wouldn't have to handle any styling at all!

Reply:
I see. We can move the label styling directly into PillButtonViewStyle without needing to create a custom LabelStyle. From my understanding, the difference between the button and the pill button is that the pill button will be used in a compound control (PillButtonBar) and has states that are tracked via an observable view model (unread dot and leading image). In this scenario it seems to me that it's more appropriate to keep it as a view struct so each button can hold on to its view model.